src/cli/self_update/test.rs

```diff
@@ -51,23 +56,21 @@ pub fn with_saved_reg_value(root: &RegKey, subkey: &str, name: &str, f: &mut dyn
 }
 
 #[cfg(windows)]
-fn get_reg_value(root: &RegKey, subkey: &str, name: &str) -> io::Result<Option<RegValue>> {
-    let subkey = root.open_subkey_with_flags(subkey, KEY_READ | KEY_WRITE)?;
-    match subkey.get_raw_value(name) {
+fn get_reg_value(root: &Key, subkey: &str, name: &str) -> io::Result<Option<String>> {
+    let subkey = root.create(subkey)?;
+    match subkey.get_string(name) {
```

Comment:
Ideally you should use `.get_value()` and `.set_value()` to replicate the functionalities of the original `get_reg_value()`.

Reply:
> @rami3l One of the function callers is the `get_path()`. I think that the paths are always string, so I directly used `get_string()`.

@InfyniteHeap In that case, I suggest renaming the functions to `(get|set)_reg_string()` for clarity.

Reply:
> > @rami3l One of the function callers is the `get_path()`. I think that the paths are always string, so I directly used `get_string()`.
> 
> @InfyniteHeap If you are sure that it's the only call site, I suggest renaming the functions to `(get|set)_reg_string()` for clarity.

I re-checked this function's callers and I'm sure that those who calls this function will only get string.